### PR TITLE
[SymbolGraph] Inherit availability from parent contexts

### PR DESCRIPTION
**Pull request by bitjammer**
Also be a little smarter about encountering duplicate `@available`
attributes on the same declaration.

rdar://63570830

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Comment by bitjammer** [comment]
@swift-ci Please smoke test Linux Platform

File: test/SymbolGraph/Symbols/Mixins/Availability/Inherited/IntroducedReplaced.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public struct S {
+  // Not possible: declarations cannot be more available than an enclosing scope
+  // @available(macOS, introduced: 10.10)
+  // public func lessThan() {}
+
+  @available(macOS, introduced: 10.16)
+  public func greaterThan() {}
+
+  @available(macOS, introduced: 10.15)
+  public func equal() {}
```
Comment (franklinsch) [inline test/SymbolGraph/Symbols/Mixins/Availability/Inherited/IntroducedReplaced.swift:19]: Can we also add tests for when a symbol's transitive parent has an availability annotation? Something like:

```swift
@available(macOS, introduced: 10.15)
public struct S {
  public struct T {
    public func foo() {}
  }
}
```

Comment (bitjammer) [reply test/SymbolGraph/Symbols/Mixins/Availability/Inherited/IntroducedReplaced.swift:19]: Ah yes, I have some tests for those but forgot to add them to this commit. Will add them.

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Pull request merged**